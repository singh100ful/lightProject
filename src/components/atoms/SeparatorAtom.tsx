import React from 'react';
import {View, StyleSheet} from 'react-native';
import {scalePresets} from '../../theme/scale';
import {ColorPresets} from '../../theme/color';

interface SeparatorAtomProps {
  height?: number;
}

export const SeparatorAtom: React.FC<SeparatorAtomProps> = ({height}) => {
  return (
    <View style={styles.container}>
      <View style={[styles.line, {height: height ? height : 1}]} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    paddingVertical: scalePresets.mediumScale,
    paddingHorizontal: scalePresets.baseScale,
  },
  line: {flex: 1, backgroundColor: ColorPresets.border},
});
