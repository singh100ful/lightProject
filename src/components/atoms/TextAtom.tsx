import * as React from 'react';
import {Text, TextProps, TextStyle} from 'react-native';
import {ColorPresets} from '../../theme/color';
import {defaultTexts} from '../../theme/fonts';

interface TextAtomProps extends TextProps {
  text: string;
  style?: TextStyle;
  preset?: TextPresets;
}

export const TextAtom: React.FC<TextAtomProps> = ({
  text,
  style,
  preset = 'body',
  ...rest
}) => {
  const textStyle = [Presets[preset], style];
  return (
    <Text
      style={[
        {
          flexShrink: 1,
          flexWrap: 'wrap',
          color: ColorPresets.primaryText,
        },
        textStyle,
      ]}
      {...rest}>
      {text}
    </Text>
  );
};

const Presets = {
  title1: {...defaultTexts.title1} as TextStyle,
  title2: {...defaultTexts.title2} as TextStyle,
  bodyBold: {...defaultTexts.bodyBold} as TextStyle,
  body: {...defaultTexts.body} as TextStyle,
  caption: {...defaultTexts.caption} as TextStyle,
};

export type TextPresets = keyof typeof Presets;
