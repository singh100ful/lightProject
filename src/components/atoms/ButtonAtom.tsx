import * as React from 'react';
import {Pressable, PressableProps, StyleSheet} from 'react-native';
import {TextAtom} from './TextAtom';
import {scalePresets} from '../../theme/scale';
import {ColorPresets} from '../../theme/color';

interface ButtonAtomProps extends PressableProps {
  title: string;
}

export const ButtonAtom: React.FC<ButtonAtomProps> = ({title, ...rest}) => {
  return (
    <Pressable style={styles.container} {...rest}>
      <TextAtom text={title} preset="title2" />
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: scalePresets.baseScale,
    paddingVertical: scalePresets.mediumScale,
    alignItems: 'center',
    backgroundColor: ColorPresets.primaryRed,
    marginVertical: scalePresets.mediumScale,
    borderRadius: scalePresets.mediumScale,
  },
});
