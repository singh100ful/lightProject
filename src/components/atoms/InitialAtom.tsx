import * as React from 'react';
import {View, StyleSheet} from 'react-native';
import {TextAtom} from './TextAtom';
import {moderateScale} from '../../theme/scale';
import {scaleFont} from '../../theme/fonts';

interface InitialAtomProps {
  initial: string;
}

export const InitialAtom: React.FC<InitialAtomProps> = props => {
  const [color, setColor] = React.useState('');

  const getRandomColor = () => {
    var letters = '0123456789ABCDEF';
    var color = '#';
    for (var i = 0; i < 6; i++) {
      color += letters[Math.floor(Math.random() * 16)];
    }
    setColor(color);
  };

  React.useEffect(() => {
    getRandomColor();
  }, []);

  return (
    <View style={[styles.container, {backgroundColor: color}]}>
      <TextAtom style={styles.initials} text={props.initial.charAt(0)} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: moderateScale(100),
    width: moderateScale(50),
    height: moderateScale(50),
  },
  initials: {color: 'white', ...scaleFont(20)},
});
