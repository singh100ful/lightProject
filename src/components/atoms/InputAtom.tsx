import * as React from 'react';
import {StyleSheet, TextInput, TextInputProps, View} from 'react-native';
import {TextAtom} from './TextAtom';
import {ColorPresets} from '../../theme/color';
import {scalePresets} from '../../theme/scale';
import {defaultTexts} from '../../theme/fonts';

interface InputAtomProps extends TextInputProps {
  label: string;
}

export const InputAtom: React.FC<InputAtomProps> = ({label, ...rest}) => {
  const [focus, setfocus] = React.useState(false);
  return (
    <View style={styles.container}>
      {label ? (
        <View style={styles.labelStyle}>
          <TextAtom
            text={label}
            style={{
              color: focus ? ColorPresets.primaryColor : ColorPresets.border,
            }}
          />
        </View>
      ) : null}
      <TextInput
        {...rest}
        onFocus={() => setfocus(true)}
        onBlur={() => setfocus(false)}
        placeholderTextColor={ColorPresets.placeholderText}
        style={[
          styles.input,
          {
            borderColor: focus
              ? ColorPresets.primaryColor
              : ColorPresets.border,
          },
        ]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {paddingVertical: scalePresets.baseScale},
  input: {
    ...defaultTexts.placeHolder,
    fontWeight: '400',
    backgroundColor: ColorPresets.white,
    borderRadius: scalePresets.smallScale,
    borderWidth: 1,
    paddingHorizontal: scalePresets.mediumScale,
    paddingVertical: scalePresets.mediumScale,
  },
  labelStyle: {
    position: 'absolute',
    left: scalePresets.mediumScale,
    top: scalePresets.smallScale,
    paddingHorizontal: scalePresets.smallScale,
    backgroundColor: ColorPresets.white,
    zIndex: 1,
  },
});
