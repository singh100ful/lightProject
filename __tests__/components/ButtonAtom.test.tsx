import React from 'react';
import renderer, {act} from 'react-test-renderer';
import {ButtonAtom} from '../../src/components/atoms/ButtonAtom';
import {Pressable} from 'react-native';

describe('Button Atom', () => {
  it('renders the passed title', () => {
    const tree = renderer
      .create(<ButtonAtom title="Click me!" onPress={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress prop when pressed', () => {
    const onPressMock = jest.fn();

    const component = renderer.create(
      <ButtonAtom title="Click me!" onPress={onPressMock} />,
    );
    const pressable = component.root.findByType(Pressable);

    act(() => {
      pressable.props.onPress();
    });

    expect(onPressMock).toHaveBeenCalled();
  });
});
