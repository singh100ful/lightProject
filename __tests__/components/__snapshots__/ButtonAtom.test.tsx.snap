// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Atom renders the passed title 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#3F5A9F",
      "borderRadius": 30.833333333333336,
      "marginVertical": 15.416666666666668,
      "paddingHorizontal": 23.125,
      "paddingVertical": 15.416666666666668,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "flexShrink": 1,
          "flexWrap": "wrap",
        },
        undefined,
      ]
    }
  >
    Click me!
  </Text>
</View>
`;
